### ISSUE TITLE
Dolphin core segmentation fault on loading

### ISSUE THREAD

**Issue by Shoegzer** [libretro/dolphin#66]
Dolphin core built under Linux Mint 19.1 x64, kernel 4.18 and GCC7 crashes with segmentation fault on loading any game.  Launching RA with verbose, I get [this](https://justpaste.it/3fa46).  

I'm wondering if this is somehow related to https://github.com/libretro/dolphin/issues/43 though it doesn't seem to be.

@twinaphex or @Spudd86 do you have any ideas here?

**Comment by i30817** [comment]
mkdir(/home/shoegazer.config/) error: Permission denied.
Failed to create directory: "/home/shoegazer.config/".
Failed to create directory: "/home/shoegazer.config/".
Failed to create directory: "/home/shoegazer.config/retroarch/".
Failed to create directory: "/home/shoegazer.config/retroarch/cheats/".


Seems like you screwed up one or more of the paths on your retroarch.cfg "shoegazer.config" -> "shoegazer/.config"

edit: or you can use "~/.config" if you're on linux like it seems. That is slightly more portable.

**Comment by Shoegzer** [comment]
Actually, the issue with the config path was related to [this](https://github.com/libretro/RetroArch/issues/7836) issue which has been fixed - but that wasn't why the dolphin core was crashing.  

With some recent update to the dolphin core, it seems that when video_context_enable=sdl_gl (as needed to fix rumble [here](https://github.com/libretro/RetroArch/issues/7426)) and video driver=gl, the core crashes.  I can fix it by removing the setting, but then I have no rumble.  Given the importance of rumble to gamecube games I don't want to run the dolphin core until this is fixed.

@twinaphex who with experience in the input subsystem might be best to look into [this](https://github.com/libretro/RetroArch/issues/7426)?  It's now causing other issues as well.  As always I'm happy to help test where needed.

**Comment by i30817** [comment]
A deadlock on startup the first time (when the core copied the dolphin files in the system dir to the savegame/dolphin-emu/Sys dir) was recently fixed but that's not likely your problem, since you mentioned a crash, not a freeze.

But i'd check anyway. https://github.com/libretro/dolphin/pull/70. The builds from the online updater already have it.

**Comment by Shoegzer** [comment]
Thanks, but again as I said, it's related to a config setting which is necessary to fix rumble on linux systems.

**Comment by i30817** [comment]
Excuse me, i misread it.

**Comment by gouchi** [comment]
Hi @Shoegzer,

Is it still an issue ? As the core has been updated recently to latest Dolphin upstream.

Thank you.

**Comment by Shoegzer** [comment]
@gouchi Thanks for checking in on this one. Indeed it works fine now, closing.

**Comment by Shoegzer** [comment]
One more comment, this is based on a core build from May 14.  I haven't tested more recent commits yet but I assume they're fine too.  

That said, Dolphin upstream currently has an [issue](https://bugs.dolphin-emu.org/issues/13888) that prevents building on Linux with the Vulkan 1.4 API installed, and I haven't checked to see whether the RetroArch core inherits it or not.  @gouchi did you have the Vulkan 1.4 API installed when you last built the core?

**Comment by gouchi** [comment]
@Shoegzer Yes I tested using Nvidia driver Vulkan 1.4 but I don't know which context Dolphin Libretro core is asking.

**Comment by Shoegzer** [comment]
Okay, then the issue must be specific to upstream.  Thanks.
